dotnet/src/Planners/Planners.OpenAI/Stepwise/FunctionCallingStepwisePlanner.cs

```diff
@@ -75,7 +80,11 @@ public sealed class FunctionCallingStepwisePlanner
         // Create and invoke a kernel function to generate the initial plan
         var initialPlan = await this.GeneratePlanAsync(question, clonedKernel, logger, cancellationToken).ConfigureAwait(false);
 
-        var chatHistoryForSteps = await this.BuildChatHistoryForStepAsync(question, initialPlan, clonedKernel, promptTemplateFactory, cancellationToken).ConfigureAwait(false);
+        if (chatHistoryForSteps is null)
+        {
+            // Build chat history for the first step
+            chatHistoryForSteps = await this.BuildChatHistoryForStepAsync(question, initialPlan, clonedKernel, promptTemplateFactory, cancellationToken).ConfigureAwait(false);
+        }
```

Comment:
I like the idea of providing a `ChatHistory` to be used inside the planner but I'm a bit worried that the effect of providing it in `InvokePlanAsync` will not be clear to people. As it is now, providing a `ChatHistory` causes the initial planning step to be ignored/skipped which is totally valid for the scenario you describe when you're continuing on from a previous planner run. However, I could see this also being used to provide more context to the planner for a first run, in which case you would likely still want the initial planning step to happen.

Reply:
Thanks, I got what you meant, so we might want to append the context in addition to the initial plan steps in some cases. It seems like we can have another flag to indicate this. But do you think it makes sense to keep it for future PR when needed?